server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2009,6 +2009,7 @@ public void forceMerge(
             throw ex;
         } catch (Exception e) {
             try {
+                indexWriter.flush();
```

Comment:
`flush()` calls internally `flush(boolean triggerMerge, boolean applyAllDeletes)`. In cases where the forceMerge is failing, should we just do a `flush(false, true)` rather than `flush(true, true)`?

Reply:
```flush(false, true)``` is not a public function. We cannot call it from outside ```InternalEngine``` (Lucene).

Reply:
Oh, yes. May be let's just check and confirm that the `triggerMerge` is not triggering merge again after the temporary files clean up.

Reply:
We should move this below `maybeFailEngine("force merge", e);` . In corruption cases, flush is not going to help out .

Reply:
Flush is optional for a given force merge . Does it guarantee for clean up the temporary files ?

Reply:
As discussed offline, ```InternalEngine``` ```flush()``` is different from ```IndexWriter``` ```flush()```. Deletion of temporary new files is happening inside ```IndexWriter``` ```deleteNewFiles``` which is called from ```IndexWriter``` ```flush()```.

Reply:
Addressed the comments.